### PR TITLE
Add `exclude_tags` parameter to recall API

### PR DESCRIPTION
**Pull request by abhihyder**
## Summary

Adds support for excluding specific tags in the `/recall` endpoint to enable efficient cross-conversation memory retrieval without client-side filtering.

## Motivation

When building multi-conversation chat applications, we need to recall:
- **Short-term memories**: Only from the current conversation
- **Long-term memories**: From ALL conversations EXCEPT the current one

Currently, the recall API only supports positive tag filtering. To achieve cross-conversation recall while excluding the current conversation, clients must:
1. Request more results than needed (e.g., 10 instead of 3)
2. Filter out unwanted tags client-side
3. Trim to desired limit

This is inefficient and can miss relevant memories if the current conversation dominates recent activity.

#### 1. Parameter Parsing (`automem/api/recall.py`)
- Added `exclude_tags` parameter parsing in `handle_recall()` function
- Supports comma-separated values: `?exclude_tags=conv1,conv2`
- Supports multiple parameters: `?exclude_tags=conv1&exclude_tags=conv2`

#### 2. Filter Logic (`app.py`)
- Enhanced `_result_passes_filters()` to accept and process `exclude_tags`
- Implements exclusion logic with both exact and prefix matching
- Excludes if ANY excluded tag is found (OR logic)

#### 3. Filter Application (`automem/api/recall.py`)
- Applied exclusion filter to all search paths:
  - Vector search results
  - Graph keyword search results
  - Tag-only search results
  - Relation expansion results
  - Entity expansion results

### Testing

Added comprehensive test coverage in `tests/test_api_endpoints.py`:
- `test_recall_with_exclude_tags_single` - Single tag exclusion
- `test_recall_with_exclude_tags_multiple` - Multiple tag exclusions
- `test_recall_with_exclude_tags_prefix_matching` - Prefix-based exclusion
- `test_recall_with_tags_and_exclude_tags_combined` - Combined positive and negative filtering
- `test_recall_exclude_tags_with_no_results` - Edge case handling

**Test Results:**
```
14 passed in 1.58s
✅ All existing recall tests pass
✅ All new exclude_tags tests pass
```

### Documentation

Updated `docs/API.md`:
- Added `exclude_tags` to recall endpoint parameters
- Included usage examples
- Documented behavior and matching rules

### New Parameter

**`exclude_tags`** (optional)
- Type: `string` or `string[]`
- Behavior: Excludes memories containing ANY of the specified tags
- Matching: Both exact and prefix matching supported
- Logic: OR-based (excludes if any tag matches)

### Examples

```bash
# Exclude single conversation
GET /recall?tags=user_1&exclude_tags=conversation_5&limit=3

# Exclude multiple conversations (comma-separated)
GET /recall?tags=user_1&exclude_tags=conversation_5,conversation_6&limit=5

# Exclude assistant messages from a conversation
GET /recall?tags=conversation_3&exclude_tags=assistant&limit=10

# Exclude temporary/draft memories (prefix matching)
GET /recall?query=project%20plan&exclude_tags=temp,draft&limit=10
```

### Exclusion Rules
1. **Exact Match**: Tag exactly matches an excluded tag
2. **Prefix Match**: Tag starts with an excluded tag (e.g., `temp` excludes `temp:draft`)
3. **OR Logic**: Memory is excluded if ANY excluded tag matches

### Filter Order
1. Time filtering (if specified)
2. Positive tag filtering (if specified)
3. Exclusion filtering (if specified)

## Backward Compatibility

✅ **Fully backward compatible**
- New parameter is optional (defaults to empty list)
- Existing API calls work unchanged
- No changes to response format
- All existing tests pass

## Use Cases

1. **Cross-conversation recall**: Get memories from all conversations except current
2. **Exclude low-importance**: Filter out temporary, draft, or test memories
3. **Role-based filtering**: Exclude assistant messages, only get user messages
4. **Conversation summaries**: Build summaries excluding meta/noise tags
5. **"Forget this conversation"**: Exclude specific conversation from results

## Performance

- **Minimal overhead**: Simple set/prefix comparison
- **More efficient than client-side filtering**: No over-fetching
- **Accurate results**: Gets correct memories even when excluded items dominate

## Related Issues

Resolves https://github.com/verygoodplugins/automem/issues/57

## Checklist

- [x] Implementation complete
- [x] All existing tests pass
- [x] New tests added with full coverage
- [x] Documentation updated
- [x] API examples provided
- [x] Backward compatible
- [x] No breaking changes
- [x] Code follows existing patterns
- [x] Type hints maintained

### CODE REVIEW

File: tests/test_api_endpoints.py [hunk 1]
```diff
@@ -1095,3 +1095,194 @@ def test_rate_limiting(client, mock_state, auth_headers):
...
+    
+    # Exclude the only conversation
+    response = client.get(
+        "/recall?tags=user_1&exclude_tags=conversation_5&limit=10",
+        headers=auth_headers
+    )
+    assert response.status_code == 200
+    data = response.get_json()
+    assert data["status"] == "success"
+    assert len(data.get("memories", [])) == 0
```
[the remark this answers is not in the document]
Comment (jack-arturo) [reply tests/test_api_endpoints.py]: Fixed in b5fd4ff - used `@pytest.mark.usefixtures("mock_state")` to silence ARG001, and the `/store` → `/memory` + `memories` → `results` fix was already applied in a prior commit.

**Comment by jack-arturo** [comment]
Thanks @abhihyder !! 💛🙏
